pkg/metrics/series.go

```diff
@@ -433,9 +461,12 @@ func marshalSplitCompress(iterator serieIterator, bufferContext *marshaler.Buffe
 		}
 	}
 
-	err = finishPayload()
-	if err != nil {
-		return nil, err
+	// if the last payload has any data, flush it
+	if seriesThisPayload > 0 {
```

Comment:
What do you think about moving the condition `if seriesThisPayload > 0` into `finishPayload`, so that `finishPayload` creates a payload only if it is not empty? It will not change the behavior but I think it is more natural that `finishPayload` handles this check.

Reply:
Sounds good.